src/main/java/uk/ac/sanger/sccp/stan/service/work/WorkServiceImp.java

```diff
@@ -72,6 +77,9 @@ public WorkWithComment updateStatus(User user, String workNumber, Status newStat
         WorkEvent event = workEventService.recordStatusChange(user, work, newStatus, commentId);
         work.setStatus(newStatus);
         String commentText = (event.getComment()==null ? null : event.getComment().getText());
+        if (work.getPriority()!=null && work.isClosed()) { // note: work has new status so isClosed() will work
+            work.setPriority(null);
+        }
```

Comment:
Just to confirm- do we need to reset the priority value when status changes to 'Closed' or 'Failed'?